Added * in every perturbation function

priyanshurohilla opened the pull request:
Addresses #935

astrojuanlu commented:
@priyanshurohilla Thanks for opening this PR. I feel the need to make a clarification about the dynamics of the project though, because perhaps the context looks evident to me, but it isn't evident for everybody else.

We are having a _conversation_ in #935, where Greg opened a feature request (and in fact, the "RFC" part of the title means "Request For Change"). As stated [in our core principles](https://github.com/poliastro/poliastro/wiki/Core-principles), *"decisions are taken in group"*. Therefore, I asked Greg a clarification about a part of the problem that I didn't understand, then he replied with some detail, and I made a **proposal** for a solution. So far, so good.

However, the fact that you didn't wait for Greg's answer to make this change is distracting and discouraging to me in several ways:

* There's a possibility that I didn't fully grasp the problem yet, in which case this solution could be wrong, or incomplete, or misleading.
* Even if the solution is correct (i.e. adding a `*` to a few function calls) there is a chance I wanted to label that as a ["good first issue"](https://github.com/poliastro/poliastro/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22), of which we have none open at the moment, to encourage new contributors to open their first pull request. But you didn't allow me to do so.
* Related to my previous point, you have been contributing to the project [at least since January 31st](https://github.com/poliastro/poliastro/pull/838) and monitoring the issue tracker for three months already, so I would have expected you to absorb these "norms" of netiquette, some of which are explicit (see the core principles above), some of which are implicit (but nonetheless stated by me in this issue tracker several times). Yes, you agreed to not say "I will work on this" on issues, like I asked you to do. But in my view, that's a tiny step, and I would have expected more.

You might feel compelled to contribute trivial features, because you have left comments in several issues asking for clarifications about how to develop more complex stuff and I haven't answered yet. There are a lot of reasons for me not answering: maybe I don't know yet, maybe I'm focusing my energy in other things (both inside and outside of the project), maybe I want others to chime in, maybe I don't feel like answering something for whatever reason.

Should I say "I am sorry" for this? I am honestly not sure. There's one thing I know for sure though: **I won't force myself to answer all the comments that happen in this project, no matter how legitimate and well-intentioned they are, because poliastro is an open source project I do voluntarily for fun**. Therefore, if there's no way I will force myself to answer *everything*, I feel that apologizing for not doing it is useless and dishonest.

Writing this wall of text is already difficult for me. In fact, I'm literally shaking. But it was also difficult to call you out in other issues several times, it was difficult to see how you engaged in a "competition" with other prospective students to submit a pull request faster, it was difficult yesterday to merge a pull request that had pending comments because you refused to address them, and in general it is very draining to face these small problems for the first time in the history of the project.

It's likely that I am overreacting. After all, it's only a few `*` here and there, what's the harm? But I hope you understand that this situation has been building up with time, and that you acknowledged that I tried to be very constructive about your behavior in the past.

As a result, I am going to close this pull request and ask you to take a few days to reflect on how do you want to interact with the project, so that everybody can keep working on polisatro while having a positive interaction with each other, learning, and having fun.

priyanshurohilla commented:
@astrojuanlu  I just want to say sorry for what all happened, but i really didn't had any intentions for that. I just wanted to contribute more and more to this project as i find it very interesting and very accurate for astrodynamics that's why sometimes i become enthusiastic to make some changes in the codebase and have some contributions to this beautiful project. I really love poliastro and i had my nights gone just working on it till i solve a particular issue, because that's my passion and i want to do it.

I didn't know these steps of mine were creating some problems in the overall workflow of the project. It was just that this was my way of showing more love to the project. I am really sorry what all i have done , i genuinely didn't know that these steps of mine were causing a problem.

I appreciate your concern and really thank you for writing this msg for me as this will guide me on how to proceed with poliastro.

Also regarding opening the PR for this issue, I had in my mind that whatever further changes needed to be made could be done in the PR as i was willing to do any kind of changes. As stated by you i was ready for any grp discussion , it was just that i thought further discussion could be done in the PR. From now onwards I will have that thing in my mind .

Regarding your answers to some of my clarifications, i didn't say anything regarding that to you. I understand that you have so much other work to do and whenever you will get time you will answer them, So i don't spam regarding that . Also if i am working on some issue, and my progress is getting stuck because of clarification then after some days i ask for your comments, there also i understand you will give the answer whenever you will get time, but i do so because if you forget, i can remind you.

Not answering the clarifications is completely normal and i understand that and you dont have to be sorry. I understand that thing.

Regarding yesterday's PR i was experimenting something on my pc which messed up the codebase and git. So i thought after having that merged i can start again from there by updating my local repo , but soon after the PR was merged i learned the `reset` command. I don't ask for several things in the chat because i like to solve them myself without bothering any of my mentors and i didn't refuse to address them, it was just that i would be addressing all those issues in the next PR.

At last @astrojuanlu  i would like to say that now i have read and understood all the different guidelines, core principles and project workflow of poliastro and i am really sorry for my behavior which i also didn't know was creating problems and really appreciate that you took out your time to make me understand where i was going wrong.

From now onwards i"ll be contributing more important stuff to the poliastro with more sincerity and dedications without creating any such problems😊

Also just a quick FYI groundtrack plotter is almost complete and will be discussing further on it in the issue tracker😀

And again thank you so much for letting me know🚀
